optaplanner-core/src/main/java/org/optaplanner/core/impl/score/ScoreUtils.java

```diff
@@ -118,7 +118,7 @@ public static double calculateTimeGradient(Number[] totalDiffNumbers, Number[] s
                 // timeGradient += 0.0
                 break;
             } else {
-                double levelTimeGradient = (double) scoreDiffLevel / (double) totalDiffLevel;
+                double levelTimeGradient = scoreDiffLevel / totalDiffLevel;
```

Comment:
-1 veto. This is a bug. Given scorediffLevel 4 and totalDiffLevel 5, the levelTimeGradient changes from 0.8 to 0.0.

Reply:
nevermind, it isn't.

Reply:
Not true. Both of those values are already doubles, the casts do nothing.